audobject/core/utils.py

```diff
@@ -131,12 +192,13 @@ def is_class(value: typing.Any):
     return False
 
 
-def split_key(key: str) -> [str, str, typing.Optional[str]]:
-    r"""Split value key in module, class and version."""
+def split_key(key: str) -> [str, str, str, typing.Optional[str]]:
+    r"""Split value key in package, module, class and version."""
     version = None
     if define.VERSION_TAG in key:
         key, version = key.split(define.VERSION_TAG)
     tokens = key.split('.')
     module_name = '.'.join(tokens[:-1])
     class_name = tokens[-1]
-    return module_name, class_name, version
+    package_name = module_name.split('.')[0]
```

Comment:
So basically, it will not work for `audtorch-internal` as the module name is `audtorch_internal`, right?
As there might be several packages using `-`, which can never be used as a module name, maybe it is safe to simply replace `_` with `-`. But the problem is here we need to deal with it the other way around, and there could be more complicated packages like `sphinxcontrib-katex` (package name) with corresponding module name `sphinxcontrib.katex`.

So I guess we need to provide the possibility to provide package names for modules as an argument, and catch an the error during installation and point the user to the possibility to provide these alias names.

Reply:
> maybe it is safe to simply replace _ with -

Yes that might be a good idea

> So I guess we need to provide the possibility to provide package names for modules as an argument,

I would rather extend the YAML format to also store the package name if it differs from the package name. With Python 3.8 there is now a possibility to do that: https://docs.python.org/3/library/importlib.metadata.html#module-importlib.metadata 

But I would actually propose to do that in a separate PR as it will anyway not be backward compatible. So maybe for the moment we simply add a note that it only works if the module name matches the package name?

Reply:
Yes, please open an issue for storing the package name, and here we continue with the current solution, but add a comment to the docstring.

Reply:
https://github.com/audeering/audobject/issues/36

Reply:
I add a comment to the docstring. But after a second thought, it might be dangerous to replace `_` with `-` as the package name could be called `audtorch_internal`. We could of course give it a try if the package cannot be installed, but maybe we should simply also rely on #36.

Reply:
No, I would not automatically replace `_` with `-`. Just stay with the approach of doing nothing.

Reply:
Ok, so I think there is nothing left to do then for the moment, right?

Reply:
Right.